### PR TITLE
Change Requests for #2475

### PR DESCRIPTION
**Pull request by StephenHodgson**
Overview
---
Change Requests for https://github.com/Microsoft/MixedRealityToolkit-Unity/pull/2475

### CODE REVIEW

**Comment by StephenHodgson** [comment]
This isn't quite ready yet, but I figured you'd have some counter arguments so fire away 🔫

**Comment by StephenHodgson** [comment]
Okay, it's as ready as it'll ever be. bang bang.

**Comment by SimonDarksideJ** [comment]
Needs a fresh pull to align PR.  Consumed as many of the changes that I didn't have issue with.
Also applied some of the standards update to multiple areas.

File: Assets/MixedRealityToolkit/_Core/Definitions/Devices/SupportedControllerType.cs [hunk 1]
```diff
@@ -10,11 +10,10 @@ namespace Microsoft.MixedReality.Toolkit.Internal.Definitions.Devices
     public enum SupportedControllerType
     {
         None = 0,
-        MixedRealityMotionController,
-        HoloLensClicker,
+        MixedRealityController,
         GenericOpenVR,
         ViveWand,
-        ViveKnuckles,
+        ValveKnuckles,
```
[the remark this answers is not in the document]
Comment (StephenHodgson) [reply Assets/MixedRealityToolkit/_Core/Definitions/Devices/SupportedControllerType.cs]: The clicker is just another mixed reality controller

File: Assets/MixedRealityToolkit/_Core/Definitions/Devices/MixedRealityControllerMapping.cs [hunk 2]
```diff
@@ -91,5 +103,31 @@ public bool DefaultModel
         [SerializeField]
         [Tooltip("Details the list of available buttons / interactions available from the device.")]
         private MixedRealityInteractionMapping[] interactions;
+
+        /// <summary>
+        /// Sets the default interaction mapping based on the current controller type.
+        /// </summary>
+        public void SetDefaultInteractionMapping()
+        {
+            if (controller.Type == null)
```
[the remark this answers is not in the document]
Comment (StephenHodgson) [reply Assets/MixedRealityToolkit/_Core/Definitions/Devices/MixedRealityControllerMapping.cs]: It's better than that mapping library monstrosity 😋

File: Assets/MixedRealityToolkit/_Core/Devices/OpenVR/OculusTouchController.cs [hunk 3]
```diff
@@ -42,30 +44,22 @@ public override void UpdateController(XRNodeState xrNodeState)
...
-                    case DeviceInputType.Grip:
                     case DeviceInputType.GripPress:
                     case DeviceInputType.GripTouch:
                         UpdateGripData(Interactions[i]);
                         break;
                     case DeviceInputType.ThumbStick:
-                    case DeviceInputType.ThumbStickTouch:
-                    case DeviceInputType.ThumbStickPress:
                     case DeviceInputType.ThumbStickNearTouch:
+                    case DeviceInputType.ThumbNearTouch:
```
[the remark this answers is not in the document]
Comment (StephenHodgson) [reply Assets/MixedRealityToolkit/_Core/Devices/OpenVR/OculusTouchController.cs]: The only difference between primary and secondary is that it's left and right handed.

The actions shouldn't be the same, I'll double check that they're separate.

[the remark this answers is not in the document]
Comment (StephenHodgson) [reply Assets/MixedRealityToolkit/_Core/Devices/OpenVR/OculusTouchController.cs]: 1 is most likely a typo

File: Assets/MixedRealityToolkit/_Core/Devices/OpenVR/OpenVRDeviceManager.cs [hunk 4]
```diff
@@ -141,22 +141,22 @@ private SupportedControllerType CurrentControllerType
                         case "Vive Wand - Left":                                                // TODO: Yet to test
                         case "Vive Wand - Right":                                               // TODO: Yet to test
                             return SupportedControllerType.ViveWand;
-                        case "Vive Knuckles - Left":                                            // TODO: Yet to test
-                        case "Vive Knuckles- Right":                                            // TODO: Yet to test
-                            return SupportedControllerType.ViveKnuckles;
-                        default:
-                            break;
+                        case "Valve Knuckles - Left":                                            // TODO: Yet to test
```
[the remark this answers is not in the document]
Comment (StephenHodgson) [reply Assets/MixedRealityToolkit/_Core/Devices/OpenVR/OpenVRDeviceManager.cs]: Just going with what is definedon unity open VR input mapping page: https://docs.unity3d.com/Manual/OpenVRControllers.html

File: Assets/MixedRealityToolkit/_Core/Devices/OpenVR/OpenVRDeviceManager.cs [hunk 5]
```diff
@@ -141,22 +141,22 @@ private SupportedControllerType CurrentControllerType
...
+                        case "Valve Knuckles - Right":                                            // TODO: Yet to test
+                            return SupportedControllerType.ValveKnuckles;
+                        case "Oculus Remote":
+                            return SupportedControllerType.OculusRemote;
                     }
                 }
 
-                return returnControllerType;
+                return SupportedControllerType.None;
```
[the remark this answers is not in the document]
Comment (StephenHodgson) [reply Assets/MixedRealityToolkit/_Core/Devices/OpenVR/OpenVRDeviceManager.cs]: Is WMR supported by open VR input?

File: Assets/MixedRealityToolkit/_Core/Devices/OpenVR/OculusTouchController.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (StephenHodgson) [reply Assets/MixedRealityToolkit/_Core/Devices/OpenVR/OculusTouchController.cs]: I see what happened. I didn't get to clean up this update loop.

**Comment by SimonDarksideJ** [comment]
Well, they're your changes.  There was only one part I didn't take, which was the data gathering from each input

File: Assets/MixedRealityToolkit/InputSystem/MixedRealityInputManager.cs [hunk 6]
```diff
@@ -868,6 +868,16 @@ public void RaiseOnInputPressed(IMixedRealityInputSource source, MixedRealityInp
             HandleEvent(floatInputEventData, OnInputPressedEventHandler);
         }
 
+        /// <inheritdoc />
+        public void RaiseOnInputPressed(IMixedRealityInputSource source, Handedness handedness, MixedRealityInputAction inputAction)
```
Comment (SimonDarksideJ) [inline Assets/MixedRealityToolkit/InputSystem/MixedRealityInputManager.cs]: Is this supposed to be in this PR?

Comment (StephenHodgson) [reply Assets/MixedRealityToolkit/InputSystem/MixedRealityInputManager.cs]: Yes it is.

File: Assets/MixedRealityToolkit/_Core/Definitions/Devices/SupportedControllerType.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (StephenHodgson) [reply Assets/MixedRealityToolkit/_Core/Definitions/Devices/SupportedControllerType.cs]: Why, we don't need it. It's covered by all the existing WMR definitions.  Same with Hand input.

**Pull request closed without merging**